packages/tools/src/utilities/math/polyline/getLineSegmentIntersectionsIndexes.ts

```diff
@@ -0,0 +1,30 @@
+import type { Types } from '@cornerstonejs/core';
+import lineSegmentsIntersect from './lineSegmentsIntersect';
+
+/**
+ * Get all intersections between a polyline and a line segment.
+ */
+
+export default function getLineSegmentIntersectionsIndexes(
+  polyline: Types.Point2[],
+  p1: Types.Point2,
+  q1: Types.Point2,
+  closed = true
+): Types.Point2[] {
+  const intersections: Types.Point2[] = [];
+  const numPoints = polyline.length;
+  const maxI = numPoints - (closed ? 1 : 2);
+
+  for (let i = 0; i <= maxI; i++) {
+    const p2 = polyline[i];
+    // Do not use % operator for better performance
+    const j = i === numPoints - 1 ? 0 : i + 1;
```

Comment:
same comment (to handle the last -> first) outside to remove condition inside the loop